File: 24_06_2024_Minimum_Number_of_K_Consecutive_Bit_Flips.cpp
class Solution {
public:
    int minKBitFlips(vector<int>& nums, int k) {
        int n = nums.size();
        vector<int> flipped(n, 0);
        int times = 0, ans = 0;
        for (int i = 0; i < n; i++) {
            if (i >= k) {
                times -= flipped[i - k];
            }
            if (((nums[i] == 1 && times % 2 == 1) || (nums[i] == 0 && times % 2 == 0))) {
                if (i + k > n) {
                    return -1;
                }
                ans++;
                times++;
                flipped[i] = 1;
            }
        }
        return ans;
    }
};

File: 24_06_2024_Minimum_Number_of_K_Consecutive_Bit_Flips.py
class Solution(object):
    def minKBitFlips(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        flipped=[0]*len(nums)
        times=ans=0
        for i in range(len(nums)):
            if i>=k:
                times-=flipped[i-k]
            if ((nums[i]==1 and times%2==1) or (nums[i]==0 and times%2==0)):
                if i+k>len(nums):
                    return -1
                ans+=1
                times+=1
                flipped[i]=1
        return ans
